### PR TITLE
Add aria-live announcement to notify users of the route change

### PR DESCRIPTION
**Pull request by jeferson-sb**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [X] Optimization
- [ ] Documentation Update

## Description

Add a live region to page content to indicate when the route is changed

## Related Tickets & Documents

Related to #14001 

## QA Instructions, Screenshots, Recordings

Turn on the screen reader, typing something into the search bar and the first result should be announced 

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://docs.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

### CODE REVIEW

**Review by aitchiss — COMMENTED** [review]
As the search result change results in a new page being loaded (since params are in the URL), I think it would be more beneficial to make this applicable to route changes generally, rather than just the search results (that would also allow us to consider #14001 closed and completed 🙂 )

I'd suggest having an empty `div` with the `aria-live` attribute inside the `#page-content-inner` container (found in `app/views/layouts/application.html.erb`. We can add the class `screen-reader-only` to make sure it's not visible.

Then in the page change handling code (in `app/assets/javascripts/base.js.erb` from line 125), we could inject text (the page title) into that div when the page has changed.

What do you think? It would mean we only have to maintain this code in one place, and it would solve the broader issue of route changes not being announced 🙂

File: app/assets/javascripts/base.js.erb [hunk 1]
```diff
@@ -132,6 +132,7 @@ var instantClick
           skipLink.focus();
         }
       }
+      document.getElementById('page-route-change').textContent = `Navigated to ${document.title}`;
```
Comment (aitchiss) [inline app/assets/javascripts/base.js.erb:135]:
```suggestion
      document.getElementById('page-route-change').textContent = title;
```
nitpick (blocking) - We should use the `title` variable that's already been initialized in this function, as sometimes the document title hasn't changed yet (see lines 164 onwards). Also if we remove 'navigated to...', it's more similar to default screen reader behaviour, so hopefully should be a better UX for users

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -92,6 +92,7 @@
     </div>
   <% end %>
   <div id="page-content-inner">
+    <div id="page-route-change" class="screen-reader-only" aria-live="assertive" aria-atomic="true"></div>
```
Comment (aitchiss) [inline app/views/layouts/application.html.erb:95]:
```suggestion
    <div id="page-route-change" class="screen-reader-only" aria-live="polite" aria-atomic="true"></div>
```
nitpick (blocking) - just a small tweak but we can make this polite so it doesn't interrupt any other important announcements 🙂

**Review by aitchiss — CHANGES_REQUESTED** [review]
This is looking great! I have a couple of very small change requests, but I tested this locally and it's all working great - nice work! 🙌

**Review by rhymes — APPROVED** [review]

**Review by aitchiss — APPROVED** [review]
Yay this is working great! I tested with VoiceOver and Safari - the skip link was announced, followed by the new page title. Tested a bunch of pages both logged in and out, and all seems great. Nice one ✨

**Pull request merged**